packages/client/src/features/accessibility/keyboard-tool-palette/keyboard-tool-palette.ts

```diff
@@ -212,29 +212,42 @@ export class KeyboardToolPalette extends ToolPalette {
 
     protected override createHeaderTools(): HTMLElement {
         this.headerToolsButtonMapping.clear();
+        let mappingIndex = 0;
 
         const headerTools = document.createElement('div');
         headerTools.classList.add('header-tools');
 
         this.defaultToolsButton = this.createDefaultToolButton();
-        this.headerToolsButtonMapping.set(0, this.defaultToolsButton);
+        this.headerToolsButtonMapping.set(mappingIndex++, this.defaultToolsButton);
         headerTools.appendChild(this.defaultToolsButton);
 
         this.deleteToolButton = this.createMouseDeleteToolButton();
-        this.headerToolsButtonMapping.set(1, this.deleteToolButton);
+        this.headerToolsButtonMapping.set(mappingIndex++, this.deleteToolButton);
         headerTools.appendChild(this.deleteToolButton);
 
         this.marqueeToolButton = this.createMarqueeToolButton();
-        this.headerToolsButtonMapping.set(2, this.marqueeToolButton);
+        this.headerToolsButtonMapping.set(mappingIndex++, this.marqueeToolButton);
         headerTools.appendChild(this.marqueeToolButton);
 
         this.validateToolButton = this.createValidateButton();
-        this.headerToolsButtonMapping.set(3, this.validateToolButton);
+        this.headerToolsButtonMapping.set(mappingIndex++, this.validateToolButton);
         headerTools.appendChild(this.validateToolButton);
 
+        if (this.gridManager) {
```

Comment:
Yeah, i noticed that one too, this is really not the nicest way of doing things. Although adopters may implement their own toolbar to avoid the default look, contributing toolbar items dynamically would be a great idea. I added https://github.com/eclipse-glsp/glsp/issues/1342 to track this.